kratos/utilities/derivatives_extension.h

```diff
@@ -0,0 +1,166 @@
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetZeroDerivativesDofsVector(std::size_t NodeId,
+                                              std::vector<Dof<double>::Pointer>& rVector,
+                                              ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetZeroDerivativesDofsVector method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetFirstDerivativesDofsVector(std::size_t NodeId,
+                                               std::vector<Dof<double>::Pointer>& rVector,
+                                               ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetFirstDerivativesDofsVector method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetSecondDerivativesDofsVector(std::size_t NodeId,
+                                                std::vector<Dof<double>::Pointer>& rVector,
+                                                ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetSecondDerivativesDofsVector method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetZeroDerivativesVariables(std::vector<VariableData const*>& rVariables,
+                                             ProcessInfo& rCurrentProcessInfo) const
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetZeroDerivativesVariables method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetFirstDerivativesVariables(std::vector<VariableData const*>& rVariables,
+                                              ProcessInfo& rCurrentProcessInfo) const
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
+                           "Calling DerivativesExtension base class "
+                           "GetFirstDerivativesVariables method. Please "
+                           "implement it in derrived class.",
+                           "");
+    }
+
+    virtual void GetSecondDerivativesVariables(std::vector<VariableData const*>& rVariables,
+                                               ProcessInfo& rCurrentProcessInfo) const
+    {
+        KRATOS_THROW_ERROR(std::runtime_error,
```

Comment:
please don't use the old macros any more

Reply:
done :+1: